### DISCUSSION TITLE
4D Support

### DISCUSSION THREAD

**Discussion by charlie-says** [jwagner/simplex-noise.js#61] [Q&A] [answered]
I'm not sure if I've completely misunderstood, but, I thought the 4d noise could be used  to create a looping 'map' in both x & y directions? 

I seem completely unable to set the z & w values in a way that loops the noise, can anyone offer me a little guidance?

**Comment by jwagner** [comment, marked as answer]
https://codepen.io/jwagner/pen/MWPNMwL here is an example of creating a '2d looping map' using 4d noise. In short you convert your x and y coordinates into angles alpha and beta. Then you derive x and y from alpha and y and w from beta using cos/sin, converting from polar to cartesian coordinates.

Thinking about getting a 1d loop out of a 2d map is a bit less brainfucky. In that case the 1d loop would just be a circle on the 2d map. :)

I hope that helps.
